endpoints/s3/src/putobject.cpp

```diff
@@ -402,33 +640,82 @@ void irods::s3::actions::handle_putobject(
 	}
 
 	if (special_chunked_header) {
+		bool keep_dstream_open_flag = false;
 		using irods_default_transport = irods::experimental::io::client::default_transport;
 
 		// create an output file stream to iRODS - wrap all structs in shared pointers
-		// since these objects will persist than the current routine
+		// since these objects will persist longer than the current routine
 		auto tp = std::make_shared<irods_default_transport>(*conn);
 		auto d = std::make_shared<irods::experimental::io::odstream>();
 
 		// posix file stream for writing parts locally - wrapped in a shared pointer
 		// since this will persist longer than the current routine
 		std::shared_ptr<std::ofstream> ofs = std::make_shared<std::ofstream>();
 
-		if (upload_part) {
+		if (upload_part && know_part_offset) {
+			{
+				std::lock_guard<std::mutex> guard(part_shmem::part_size_map_mutex);
```

Comment:
I don't see the `part_size_map` in here. Is this mutex also protecting the `replica_token_and_number_map`?

Reply:
Yes it is.  I changed the mutex name to `multipart_global_state_mutex`.